components/core/tests/test-NetworkReader.cpp

```diff
@@ -188,3 +208,47 @@ TEST_CASE("network_reader_illegal_offset", "[NetworkReader]") {
     size_t pos{};
     REQUIRE((clp::ErrorCode_Failure == reader.try_get_pos(pos)));
 }
+
+TEST_CASE("network_reader_with_http_header_kv_pairs", "[NetworkReader]") {
```

Comment:
```suggestion
// NOLINTNEXTLINE(readability-function-cognitive-complexity)
TEST_CASE("network_reader_with_http_header_kv_pairs", "[NetworkReader]") {
```
We can safely silent this warning since it's mostly caused by catch2 macros